src/utils/database.js

```diff
@@ -88,15 +88,15 @@ function connect() {
 
 	const mongooseOptions = {
 		autoIndex: NODE_ENV !== ENVIRONMENTS.PRODUCTION,
-		poolSize: MONGOOSE_CONNECTION_POOL_SIZE,
+		maxPoolSize: MONGOOSE_CONNECTION_POOL_SIZE, // https://mongoosejs.com/docs/migrating_to_6.html#mongodb-driver-40
 		useNewUrlParser: true,
-		useFindAndModify: false,
-		useCreateIndex: true,
-		useUnifiedTopology: true,
 	};
 
 	addAuthenticationToMongooseOptions(options.username, options.password, mongooseOptions);
-
+	// https://mongoosejs.com/docs/6.x/docs/migrating_to_6.html#strictpopulate
+	mongoose.set('strictPopulate', false);
+	// https://mongoosejs.com/docs/6.x/docs/migrating_to_6.html#strictquery-is-removed-and-replaced-by-strict
+	mongoose.set('strictQuery', false);
```

Comment:
Meine Begreisterung dafür hält sich in Grenzen, aber vereinfacht einiges.

Reply:
https://mongoosejs.com/docs/7.x/docs/migrating_to_7.html#strictquery